nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/hive/streaming/NiFiRecordSerDe.java

```diff
@@ -187,132 +202,127 @@ private Object extractCurrentField(Record record, RecordField field, TypeInfo fi
                 }
                 switch (primitiveCategory) {
                     case BYTE:
-                        Integer bIntValue = record.getAsInt(fieldName);
-                        val = bIntValue == null ? null : bIntValue.byteValue();
+                        Integer bIntValue = DataTypeUtils.toInteger(fieldValue, field.getDataType().getFormat());
```

Comment:
I see that the direct call to `DataTypeUtils` is pretty much equivalent to the `record.getAs` calls (because of the implementation of `MapRecord` for example), but I personally prefer the use of the `Record` interface since we're working with Record objects and it's possible (however unlikely) that the underlying implementation will change. Unfortunately that may also mean that `getAs` returns `null` (although it apparently doesn't for primitive types) which is why the extra checks are in there. I can appreciate the slight performance increase but I'd prefer we keep the current use of `Record` methods.

Reply:
[AvroTypeUtil](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-avro-record-utils/src/main/java/org/apache/nifi/avro/AvroTypeUtil.java) does seem to use `DataTypeUtil` rather than use the interface directly so was following that and not assuming the `MapRecord` implementation.